bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/DahuaHandler.java

```diff
@@ -169,18 +169,100 @@ private void processEvent(String content) {
+                }
+                break;
+            case "ProfileAlarmTransmit":
+                if ("Start".equals(action)) {
+                    if (content.contains("DoorMagnetism"))
+                        ipCameraHandler.logger.debug("Camera Magnetic lock fail warning");
+                }
+                if ("Stop".equals(action)) {
+                    if (content.contains("DoorMagnetism"))
+                        ipCameraHandler.logger.debug("Camera Magnetic lock fail warning gone");
+                } else {
+                    ipCameraHandler.logger.debug("Unrecognised Alarm Dahua event, content={}", content);
+                }
+                break;
+            case "DoorStatus":
+                if ("Pulse".equals(action)) {
+                    if (content.contains("\"Relay\" : true")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.OFF);
+                        ipCameraHandler.logger.debug("Dahua door lock unlocked");
+                    } else if (content.contains("\"Status\" : \"Close\"")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_CONTACT, OnOffType.ON);
+                        ipCameraHandler.logger.debug("Dahua door contact closed");
+                    } else if (content.contains("\"Status\" : \"Open\"")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_CONTACT, OnOffType.OFF);
+                        ipCameraHandler.logger.debug("Dahua door contact opened");
+                    } else {
+                        ipCameraHandler.logger.debug("Unrecognised Door status Dahua event, content={}", content);
+                    }
+                }
+                break;
             case "AlarmLocal":
                 if ("Start".equals(action)) {
                     if (content.contains("index=0")) {
                         ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT, OnOffType.ON);
+                    } else if (content.contains("index=3")) {
+                        ipCameraHandler.setChannelState(CHANNEL_EXIT_BUTTON, OnOffType.ON);
+                        ipCameraHandler.logger.debug("Dahua exit button pressed");
                     } else {
                         ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT2, OnOffType.ON);
+                        ipCameraHandler.logger.debug("External alarm Dahua event, content={}", content);
                     }
                 } else if ("Stop".equals(action)) {
                     if (content.contains("index=0")) {
                         ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT, OnOffType.OFF);
+                    } else if (content.contains("index=3")) {
+                        ipCameraHandler.setChannelState(CHANNEL_EXIT_BUTTON, OnOffType.OFF);
+                        ipCameraHandler.logger.debug("Dahua exit button released");
```

Comment:
Remove the logging line as it gives no extra information that is not already logged in the event.log when the channel state changes. As a general rule, never log when you're changing the state of a channel as it taxes the machine running openHAB for zero benefit.